### PR TITLE
Fixing Enum with only one element being ignored

### PR DESCRIPTION
**Pull request by isamu-isozaki**
Fixes https://github.com/outlines-dev/outlines/issues/720

### CODE REVIEW

**Comment by rlouf** [comment]
Thank you for the quick fix! Would you mind adding a unit test for this?

**Comment by isamu-isozaki** [comment]
For sure! Let me figure that out

File: tests/fsm/test_json_schema.py [hunk 1]
```diff
@@ -163,6 +163,24 @@ def test_match_number(pattern, does_match):
...
+        ),
+        # Make sure strings are escaped
+        (
+            {"title": "Foo", "const": ".*", "type": "string"},
+            r'"\.\*"',
+            [('".*"', True), (r'"\s*"', False), (r'"\.\*"', False)],
+        ),
+        # Const integer
+        (
+            {"title": "Foo", "enum": 0, "type": "integer"},
```
Comment (rlouf) [inline tests/fsm/test_json_schema.py:180]: It looks like this test is breaking, did you mean `const`?

Comment (isamu-isozaki) [reply tests/fsm/test_json_schema.py]: Ah yes! Sorry forgot to test

Comment (isamu-isozaki) [reply tests/fsm/test_json_schema.py]: Fixed

**Comment by rlouf** [comment]
Thank you for contributing!

**Comment by isamu-isozaki** [comment]
My pleasure!

**Pull request merged**